src/authentication/git-auth-scheme.ts

```diff
@@ -0,0 +1,172 @@
+import * as auth0 from 'auth0-js';
+import * as Boom from 'boom';
+import { decode, verify } from 'jsonwebtoken';
+import * as jwksRsa from 'jwks-rsa';
+import fetch from 'node-fetch';
+import * as util from 'util';
+
+import * as Hapi from '../server/hapi';
+import getResponseJson from '../shared/get-response-json';
+import * as logger from '../shared/logger';
+import { sanitizeSubClaim } from './authentication-hapi-plugin';
+import { AccessToken } from './types';
+
+interface Auth0LoginResponse {
+  accessToken: string;
+  expisresIn: number;
+}
+
+export class GitAuthScheme {
+  private readonly _getSigningKey: (arg1: string) => Promise<jwksRsa.Key>;
```

Comment:
I think that if there are clearly two versions of a function or a method, let's say a private and a public one, it's not worth the effort to think of two *different* names. That would also make it less clear if the two methods are related or not. We could come up with another convention, but this is widely used.